enterprise/frontend/src/metabase-enterprise/sandboxes/confirmations.ts

```diff
@@ -18,6 +18,7 @@ export function getSandboxedTableWarningModal(
   // should prompt them that we will have to remove native access for all tables/schemas
   if (
     value === DataPermissionValue.SANDBOXED &&
+    permissions[groupId]?.[entityId.databaseId] &&
```

Comment:
Because the BE removes empty values in `graph` endpoints, the check would fail on an empty object.